zerver/actions/message_send.py

```diff
@@ -876,10 +926,25 @@ def do_send_messages(
         if send_request is not None
     ]
 
+    if not send_message_requests:  # nocoverage
+        return []
+
     # Save the message receipts in the database
     user_message_flags: dict[int, dict[int, list[str]]] = defaultdict(dict)
 
-    Message.objects.bulk_create(send_request.message for send_request in send_message_requests)
+    if idempotency_key is None:
+        Message.objects.bulk_create(send_request.message for send_request in send_message_requests)
+
+    else:
+        if len(send_message_requests) != 1:  # nocoverage
```

Comment:
This condition can't be true for API calls anyway, so I think it's not very important to have a test case